src/ResourceManager/Security/Commands.Security/Cmdlets/Alerts/SetAlerts.cs

```diff
@@ -0,0 +1,129 @@
+        [ValidateNotNullOrEmpty]
+        public string ActionType { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.ResourceId)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.InputObject, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.InputObject)]
+        [ValidateNotNullOrEmpty]
+        public PSSetAlertInputObject InputObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                var rg = ResourceGroupName;
+                var name = Name;
+                var actionType = ActionType;
+                var location = Location;
+
+                switch (ParameterSetName)
+                {
+                    case ParameterSetNames.SubscriptionLevelResource:
+                    case ParameterSetNames.ResourceGroupLevelResource:
+                        break;
+                    case ParameterSetNames.ResourceId:
+                        location = AzureIdUtilities.GetResourceLocation(ResourceId);
+                        name = AzureIdUtilities.GetResourceName(ResourceId);
+                        break;
+                    case ParameterSetNames.InputObject:
+                        name = InputObject.Name;
+                        actionType = InputObject.ActionType;
+                        rg = InputObject.ResourceGroupName;
+                        location = InputObject.Location;
+                        break;
+                    default:
+                        throw new PSInvalidOperationException();
+                }
+
+                SecurityCenterClient.AscLocation = location;
+
+                if (string.IsNullOrEmpty(rg))
+                {
+                    if (ShouldProcess(name, VerbsCommon.Set))
+                    {
+                        SecurityCenterClient.Alerts.UpdateSubscriptionLevelAlertStateWithHttpMessagesAsync(name, actionType).GetAwaiter().GetResult();
+                    }
+                }
+                else
+                {
+                    if (ShouldProcess(name, VerbsCommon.Set))
+                    {
+                        SecurityCenterClient.Alerts.UpdateResourceGroupLevelAlertStateWithHttpMessagesAsync(name, actionType, rg).GetAwaiter().GetResult();
+                    }
+                }
+            }
+            catch (CloudException)
+            {
+                try
+                {
+                    base.EndProcessing();
+                }
+                catch
+                {
+                    // Ignore exceptions during end processing
+                }
+
+                throw;
+            }
+        }
+    }
+
+    public class PSSetAlertInputObject
```

Comment:
same comment as above :)

Reply:
(the other comment was collapsed, so it'll be easier to discuss in this one)

What aspects of the object are different? It seems the only thing missing from the returned `PSSecurityAlert` type (from `Get-AzureRmSecurityAlert`) is the location property, but that can be something that the user has to provide as a parameter. If we introduce this new type that's different from the existing `PSSecurityAlert` type, then users will have to use a separate mechanism to create this object and then pass it to this cmdlet rather than just taking the result of `Get-AzureRmSecurityAlert` and piping (or passing) it to this cmdlet.

Reply:
this cmdlet only let you control the state of the alert (active, dismissed etc.)
the object that returns from the get has a lot more fields that cannot be edited and I think it will be misleading to let the user edit them and ignore them afterwards (like CompromisedEntity, DetectedTimeUtc, ReportedSeverity etc)